accepted/new-api.md

```diff
@@ -0,0 +1,1495 @@
+
+```rust
+struct Global(/* ... */);
+
+impl Clone for Global {}
+impl Copy for Global {}
+
+impl Global {
+    fn new(cx: impl AsContextMut, ty: GlobalType, val: Val) -> Result<Global>;
+    fn ty(&self, cx: impl AsContext) -> GlobalType;
+    fn get(&self, cx: impl AsContext) -> Val;
+    fn set(&mut self, cx: impl AsContextMut, val: Val) -> Result<()>;
+}
+```
+
+### `Table`
+
+```rust
+struct Table(/* ... */);
+
+impl Clone for Table {}
+impl Copy for Table {}
+
+impl Table {
+    fn new(&mut self, ty: TableType, init: Val) -> Result<Table>;
+    fn ty(&self, cx: impl AsContext) -> TableType;
+    fn get(&self, cx: impl AsContext, idx: u32) -> Option<Val>;
+    fn set(&self, cx: impl AsContextMut, idx: u32, val: Val) -> Result<()>;
+    fn grow(&self, cx: impl AsContextMut, delta: u32, val: Val) -> Result<u32>;
+    fn copy(
+        cx: impl AsContextMut,
+        dst_table: Table,
+        dst_index: u32,
+        src_table: Table,
+        src_index: u32,
+        len: u32,
+    ) -> Result<()>;
+    fn fill(&self, cx: impl AsContextMut, dst: u32, val: Val, len: u32) -> Result<()>;
+}
+```
+
+### `Memory`
+
+```rust
+struct Memory(/* ... */);
+
+impl Clone for Memory {}
+impl Copy for Memory {}
+
+impl Memory {
+    fn new(cx: impl AsContextMut, ty: MemoryType) -> Result<Memory>;
+    fn ty(&self, cx: impl AsContext) -> MemoryType;
+    fn read(
+        &self,
+        cx: impl AsContext,
+        offset: usize,
+        buffer: &mut [u8],
+    ) -> Result<(), MemoryAccessError>;
+    fn write(
+        &mut self,
+        cx: impl AsContextMut,
+        offset: usize,
+        buffer: &[u8],
+    ) -> Result<(), MemoryAccessError>;
+    fn data<'a, T>(&self, cx: StoreContext<'a, T>) -> &'a [u8]; // safe!
+    fn data_mut<'a, T>(&mut self, cx: StoreContextMut<'a, T>) -> &'a mut [u8]; // safe!
+}
+```
+
+### `Instance`
+
+```rust
+struct Instance(/* ... */);
+
+impl Clone for Instance {}
+impl Copy for Instance {}
+
+impl Instance {
+    fn new(cx: impl AsContextMut, module: &Module, args: &[Extern])
+        -> Result<Instance>;
+    async fn new_async(
+        cx: impl AsContextMut,
+        module: &Module,
+        args: &[Extern],
+    ) -> Result<Instance>;
+    fn ty(&self, cx: impl AsContext) -> InstanceType;
+    fn exports(&self, cx: impl AsContext) -> impl ExactSizeIterator<Item = Export>;
+    fn get_export(&self, cx: impl AsContext, name: &str) -> Option<Extern>;
+    fn get_func(&self, cx: impl AsContext, name: &str) -> Option<Func>;
+    fn get_typed_func<P, R>(&self, cx: impl AsContext, name: &str) -> Result<TypedFunc<P, R>>;
+    fn get_table(&self, cx: impl AsContext, name: &str) -> Option<Table>;
+    fn get_memory(&self, cx: impl AsContext, name: &str) -> Option<Memory>;
+    fn get_global(&self, cx: impl AsContext, name: &str) -> Option<Global>;
+}
+```
+
+### `Caller`
+
+```rust
+impl<'a, T> Caller<'a, T> {
+    fn context(&self) -> StoreContext<'_, T>;
+    fn context_mut(&mut self) -> StoreContextMut<'_, T>;
```

Comment:
Are these two functions doing more than what the `AsContext/Mut` trait methods provide?

Reply:
Nah they're just explicitly named so if users want to call them they don't have to import the trait. I'm not sure if this makes sense but it's always an option we can have of course!